sim-lib/src/lib.rs

```diff
@@ -498,6 +492,38 @@ pub struct Simulation {
     results: Arc<Mutex<PaymentResultLogger>>,
 }
 
+impl SimulationCfg {
+    pub fn new(
+        total_time: Option<u32>,
+        expected_payment_msat: u64,
+        activity_multiplier: f64,
+        write_results: Option<WriteResults>,
+        seed: Option<u64>,
+    ) -> Self {
+        Self {
+            total_time: total_time.map(|x| Duration::from_secs(x as u64)),
+            expected_payment_msat,
+            activity_multiplier,
+            write_results,
+            seeded_rng: MutRng::new(seed),
+            results: Arc::new(Mutex::new(PaymentResultLogger::new())),
```

Comment:
Is there a particular reason that the `results` Arc object needs to be in the `SimulationCfg` struct? I would expect the `SimulationCfg` type to be entirely user defined values that are simply simulation settings. The `PaymentResultLogger` feels more like a runtime object that would be a part of the `Simulation` type.

No technical reason why this won't work, just a thought on overall design. Feel free to ignore.

Reply:
> Is there a particular reason that the results Arc object needs to be in the SimulationCfg struct?

None, nice catch! I just copied everything over, it def belongs in `Simulation` 👍